python/cppyy/numba_ext.py

```diff
@@ -56,28 +59,50 @@ class Qualified:
     'unsigned long long'     : nb_types.ulonglong,
     'float'                  : nb_types.float32,
     'double'                 : nb_types.float64,
+    'const char*'           : nb_types.unicode_type,
 }
 
 def cpp2numba(val):
     if type(val) != str:
         # TODO: distinguish ptr/ref/byval
         return typeof_scope(val, nb_typing.typeof.Purpose.argument, Qualified.value)
+    elif val[-1] == '*':
+        return nb_types.CPointer(_cpp2numba[val[:-1]])
     return _cpp2numba[val]
 
 _numba2cpp = dict()
 for key, value in _cpp2numba.items():
-    _numba2cpp[value] = key
-# prefer "int" in the case of intc over "int32_t"
-_numba2cpp[nb_types.intc] = 'int'
+    if value in _numba2cpp:
+        _numba2cpp[value].append(key)
+        continue
+    _numba2cpp[value] = [key]
+
+
+def refactor_eigen_to_cpp(input_string):
+    pattern = r'(Eigen::Matrix<\w+,\d+,\d+,\d+),\d+,\d+>'
+    refactored_string = re.sub(pattern, r'\1>', input_string)
+    return refactored_string
 
 def numba2cpp(val):
     if hasattr(val, 'literal_type'):
         val = val.literal_type
         if val == nb_types.int64:      # Python int
             # TODO: this is only necessary until "best matching" is in place
             val = nb_types.intc        # more likely match candidate
+    elif val not in _numba2cpp:
+            if(isinstance(val, CppClassNumbaType)):
+                cppclass_type = (str(val).rstrip(')')).lstrip("CppClass(")
```

Comment:
The type should be available without string processing (e.g. from `__cpp_name__` from the original scope).

Reply:
pushed fix